### DISCUSSION TITLE
Anyone else getting "Method Promise.prototype.then called on incompatible receiver #<Promise>" with 3.3.8?

### DISCUSSION THREAD

**Discussion by n8ores** [node-config/node-config#701] [Q&A] [answered]
Anyone else getting this error when upgrading from 3.3.7 to 3.3.8? 

```
import config from "config";

let fn = async ()=>{
    await config .get("BASE_URL"); // This line throws the above exception
}
```

**Comment by lorenwest** [comment]
This is a question for the authors of the change from 3.3.7 to 3.3.8.  The [History.md](https://github.com/node-config/node-config/blob/master/History.md) file, unfortunately, was not updated to include this information.

My request is to update the History.md file to include the author of these changes so a request can go out to that contributor to look into this issue.

**Reply by markstos** [reply]
The changes in 3.3.8 are found here: https://github.com/node-config/node-config/releases/tag/v3.3.8

**Reply by lorenwest** [reply]
Oooooh that's much better than History.md.  I'll add a link to the top of History.md to point to the new release structure.  Thank you for putting this in place.

**Comment by markstos** [comment]
If there's actually a space between `config` and `.get` as in your example, that could be the issue. 

Also what does your BASE_URL config key return? Is it a promise?

If you think you've found a bug or regression,  write a test for the test suite which passes with 3.3.7 but passes with 3.3.8.

**Comment by thieleju** [comment]
Does anyone know how to fix this? I'm still unable to pass promises to asyncConfig(). Tested this code with version 3.3.8, 3.3.9 and 3.3.10 but no success. I'm using node 18.16.1

```js
const { asyncConfig } = require("config/async")

async function test() {
  await new Promise((resolve) => setTimeout(resolve, 1000))
  return "hello"
}

module.exports = {
    something: asyncConfig(test)
}
```
other file
```js
const { resolveAsyncConfigs } = require("config/async")

resolveAsyncConfigs(config).then(() => {
  require("./main.js")
}).catch((error) => {
  winston.info("Failed to resolve configs", error)
})
```
The error message is: 
```json
Method Promise.prototype.then called on incompatible receiver #<Promise> {
"stack":"TypeError: Method Promise.prototype.then called on in compatible receiver #<Promise>
    at Proxy.then (<anonymous>)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"
}
```

**Reply by thieleju** [reply, marked as answer]
Deleting node_modules with version 3.3.10 might have done the job for me, not exactly sure what fixed the error tbh. But the above code works fine now.

Edit: I was missing the ALLOW_CONFIG_MUTATIONS=true environment variable, which seems to be required.

**Comment by jdmarshall** [comment]
@thieleju Try 4.3.0 and tell me what you think:

https://github.com/node-config/node-config/wiki/Special-features-for-JavaScript-configuration-files#promises-and-async-functions
